Sources_v3/Utils/MulticastDelegate.swift

```diff
@@ -0,0 +1,27 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A wrapper object that allows attaching additional delegates and call the callbacks on all of them.
+///
+/// - Warning: ⚠️ Because `MulticastDelegate` keeps strong references to the delegates, it's strongly recommended to use
+/// an additional wrapper which will make sure the delegate instance is referenced only weakly.
+struct MulticastDelegate<WrappedDelegate> {
```

Comment:
You can use internal wrapper for weak references:

```swift
final class WeakRef<T: AnyObject> {
    weak var value: T?
    
    init(_ value: T) {
        self.value = value
    }
}
```

and skip empty values.

Reply:
This doesn't work for protocols.
```
protocol DelegateProtocol {}
let weak: WeakRef<DelegateProtocol> // -> compile error, T must be a concrete type
```